source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -253,5 +251,37 @@ MStatus MaterialXSurfaceNode::initialize()
 {
     CHECK_MSTATUS(inheritAttributesFrom(MATERIALX_NODE_TYPENAME));
 
+    MFnNumericAttribute nAttr;
+    VP2_TRANSPARENCY_ATTRIBUTE = nAttr.create(
+        mx::OgsXmlGenerator::VP_TRANSPARENCY_NAME.c_str(),
+        "vp2t",
+        MFnNumericData::kFloat,
+        0.0
+    );
+
+    CHECK_MSTATUS(nAttr.setInternal(true));
+    CHECK_MSTATUS(nAttr.setKeyable(false));
+
+    // otherwise Maya refuses to map it to the eponymous fragment input
+    CHECK_MSTATUS(nAttr.setWritable(true));
+
+    CHECK_MSTATUS(nAttr.setHidden(true));
+    CHECK_MSTATUS(nAttr.setAffectsAppearance(true));
+    CHECK_MSTATUS(addAttribute(VP2_TRANSPARENCY_ATTRIBUTE));
+
     return MS::kSuccess;
+}
+
+bool
+MaterialXSurfaceNode::getInternalValue(const MPlug& plug, MDataHandle& dataHandle)
+{
+    if (plug == VP2_TRANSPARENCY_ATTRIBUTE)
+    {
+        dataHandle.set(_materialXData && _materialXData->isTransparent() ? 0.5f : 0.0f);
```

Comment:
So, we just set the dummy attribute to 0.5 if we know that the surface is transparent. It's not used in any other way. I think this looks like a design flaw in MPxShadingNodeOverride - it's not expected to determine its own transparency internally. I think an explicit isTransparent() method similar to the one in MPxShaderOverride would be a useful addition to the API.

Reply:
Yes, that makes sense to extend the API. As long as it works for now :)